### PR TITLE
Minor targeting cleanup

### PR DESCRIPTION
**Pull request by penev92**
A couple of simple clean-up fixes to the current targeting:
- Removed `AttackFollow.CanAttack()` since it doesn't do anything that its `base.CanAttack()` doesn't. Also unlike `AttackFollow.CanAttack()`, the base method has a `self.IsInWorld` check, which comes after the removed target validity check.
- Exposed an Info field in the `DetectCloaked` trait in order to remove
  `a.Actor.Info.Traits.Get<DetectCloakedInfo>();`
  because `ActorsWithTrait<DetectCloaked>()` already gives us the needed trait and it is redundant to do another `Traits.Get<>()`
- `TargetableUnit.TargetableBy()` shouldn't check if the actor is cloaked. `cloak.IsVisible()` does that.

In theory this should have a marginally positive effect on late-game performance. Other than that, mostly cleanup stuff.

### CODE REVIEW

**Comment by pchote** [comment]
Looks like a sensible set of cleanups, but I haven't tested anything.

File: OpenRA.Mods.Common/Traits/DetectCloaked.cs [hunk 1]
```diff
@@ -13,14 +13,24 @@
...
-	public class DetectCloakedInfo : TraitInfo<DetectCloaked>
+	public class DetectCloakedInfo : ITraitInfo
 	{
 		[Desc("Specific cloak classifications I can reveal.")]
 		public readonly string[] CloakTypes = { "Cloak" };
 
 		[Desc("Measured in cells.")]
 		public readonly int Range = 5;
+
+	    public object Create(ActorInitializer init) { return new DetectCloaked(this); }
```
Comment (ScottNZ) [inline OpenRA.Mods.Common/Traits/DetectCloaked.cs]: Broken indentation

**Comment by Mailaender** [comment]
I played a test game of TD and can confirm that https://github.com/OpenRA/OpenRA/issues/3763, https://github.com/OpenRA/OpenRA/issues/5018 and https://github.com/OpenRA/OpenRA/issues/5247 are gone with the changes in this pull request (did not check if this is already fixed in bleed) and I can't spot any regressions. :+1:

**Comment by Phrohdoh** [comment]
The code changes look nice here but in a few test games in RA the AI still attacked some cloaked actors it had never seen before so I cannot confirm a fix.

**Comment by Mailaender** [comment]
You need to be sure that infantry, radars and base defense can decloak.

**Comment by penev92** [comment]
This is actually not supposed to fix anything. It's just removing some redundant code (some of which could actually potentially throw, but that's not really the point).

**Comment by huwpascoe** [comment]
:+1:

**Comment by Phrohdoh** [comment]
>
penev: Phrohdoh: can you do  #7276 while you're at it
Phrohdoh: No because matt promises a fix that is unrelated (apparently) and that fix doesn't work for me.
penev: Phrohdoh: but the point is that there are no regressions, which is the only thing that PR promises
Phrohdoh: cool, I won't do it with matt saying it fixes things it doesn't
Phrohdoh: he needs to clarify how it supposidly fixes things
Phrohdoh: at least for me to merge it.
penev: could you leave a comment then, so people don't get confused

**Comment by obrakmann** [comment]
I believe the issues Matt is referring to have been fixed as part of #7097.

**Comment by obrakmann** [comment]
:+1: Stealth behaviour in TD is unchanged, and the shellmap in RA is working as before as well.

**Comment by Mailaender** [comment]
> I believe the issues Matt is referring to have been fixed as part of #7097.

Hunt is only used for scripting. HackyAI doesn't use it so it is unlikely #7097 closed all those bug reports. They may have been false reports by people who did not understand which actors have uncloaking abilities or something that has been otherwise fixed in the meantime.

**Pull request merged**